src/witan/app/forecast.clj

```diff
@@ -301,13 +311,15 @@
    (go ;; temporary solution
      (log/info "Starting to run model: " (:model_id forecast) (:name model) (str "v" (:version model)))
      (try
-       (let [outputs   (mex/execute-model forecast model)
-             data      (into {} (->> (vec outputs)
-                                     (first)
-                                     (map #(process-output-data! % (:public? forecast)))))]
-         (log/info "Finished running model " (:model_id forecast) "-" (count data) "output(s) returned." outputs)
-         (conclude-forecast! (assoc (->Forecast forecast) :outputs data)))
-       (catch Exception e (log/error "Model" (:model_id forecast) "threw an error:" (.getMessage e) (clojure.stacktrace/print-stack-trace e)))))))
+       (let [outputs  (mex/execute-model forecast model)]
+         (if-let [error (:error outputs)]
+           (process-error! forecast error)
+           (let [data      (into {} (->> (vec outputs)
+                                         (first)
```

Comment:
Do you need to convert to vec then call first? won't `(first outputs)` wotk on it's own?